src/stream/src/executor/aggregation/distinct.rs

```diff
@@ -74,8 +74,11 @@ impl<S: StateStore> ColumnDeduplicater<S> {
             // get counts of the distinct key of all agg calls that distinct on this column
             let key = group_key.chain(row::once(datum));
             let compacted_key = CompactedRow::from(&key); // TODO(rc): is it necessary to avoid recomputing here?
-            let counts = if let Some(counts) = self.cache.get_mut(&compacted_key) {
-                counts
+
+            // TODO(yuhao): avoid this `contains`.
+            // https://github.com/risingwavelabs/risingwave/issues/9233
+            let mut counts = if self.cache.contains(&compacted_key) {
+                self.cache.get_mut(&compacted_key).unwrap()
             } else {
                 // load from table into the cache
```

Comment:
can we write this here? :thinking: 
```suggestion
    let mut cache = UNSAFE { cache };
```